### PR TITLE
drivers: usb: Adapt usb driver for B91 and B92

### PR DESCRIPTION
**Pull request by TelinkKrystian**
Update usb_dc_b91 driver to usb_dc_b9x for B91 and B92.

### CODE REVIEW

File: dts/riscv/telink/telink_b92.dtsi [hunk 1]
```diff
@@ -230,9 +230,9 @@
 		};
 
 		usbd: usbd@80100800 {
-			compatible = "telink,b91-usbd";
-			reg = <0x80100800 0x40>;
-			interrupts = <19 1>, <20 1>, <21 1>, <22 1>, <23 1>, <36 1>, <47 1>;
+			compatible = "telink,b9x-usbd";
+			reg = <0x80100800 0x50>;
```
Comment (serhiiSalamakha) [inline dts/riscv/telink/telink_b92.dtsi:234]: @TelinkKrystian
Why did you change from 0x40 to 0x50?
In the B91 datasheet, the offset is 0x00-0x3F

File: drivers/usb/device/Kconfig [hunk 2]
```diff
@@ -205,6 +205,32 @@ config USB_B91_WORK_QUEUE_STACK_SIZE
 	  for handling the events from the USBD ISR, i.e. executing endpoint
 	  callbacks and providing proper notifications to the USB device stack.
 
+config USB_TELINK_B92
```
Comment (serhiiSalamakha) [inline drivers/usb/device/Kconfig]: You can create common fields:
USB_TELINK_B9X
USB_B9X_EVT_QUEUE_SIZE
USB_B9X_WORK_QUEUE_STACK_SIZE

Comment (TelinkKrystian) [reply drivers/usb/device/Kconfig]: Done

**Comment by andriy-bilynskyy** [comment]
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| hal_telink | https://github.com/telink-semi/hal_telink/commit/60ad4bf3940d6b8b9bb6c2ab933d931e9c90b74c | https://github.com/telink-semi/hal_telink/commit/382a54822950feefcf8d748dc239c3cad875008f (telink_B92) | [telink-semi/hal_telink@60ad4bf3..382a5482](https://github.com/telink-semi/hal_telink/compare/60ad4bf3940d6b8b9bb6c2ab933d931e9c90b74c..382a54822950feefcf8d748dc239c3cad875008f) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

**Review by serhiiSalamakha — APPROVED** [review]
